### PR TITLE
Added user profile

### PR DESCRIPTION
**Pull request by EugeneVasilevski**

### CODE REVIEW

File: app/src/main/java/pro/itshark/moneysplitter/Constants.kt [hunk 1]
```diff
@@ -1,3 +1,17 @@
 package pro.itshark.moneysplitter
 
-const val BASE_URL = "https://moneytransfer.azurewebsites.net/"
\ No newline at end of file
+const val BASE_URL = "https://moneytransfer.azurewebsites.net/"
+
+object PrefKey {
```
Comment (BlaizMiko) [inline app/src/main/java/pro/itshark/moneysplitter/Constants.kt]: Change to full name

File: app/src/main/java/pro/itshark/moneysplitter/presentation/userProfile/UserModel.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        val email: String,
+        val name: String,
+        val surname: String,
+        val phoneNumber: Int,
+        val creditCardNumber: Long,
+        val balance: Float,
+        val token: String,
+        val imageUrl: String,
+        val backgroundImageUrl: String,
+        val friends: String
```
Comment (BlaizMiko) [inline app/src/main/java/pro/itshark/moneysplitter/presentation/userProfile/UserModel.kt]: Rly string?

File: app/src/main/java/pro/itshark/moneysplitter/presentation/userProfile/UserProfileViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import io.reactivex.schedulers.Schedulers
+import pro.itshark.moneysplitter.BR
+import pro.itshark.moneysplitter.domain.UserUseCases
+
+class UserProfileViewModel(private val userUseCases: UserUseCases) : BaseObservable() {
+
+    private var userModel: UserModel? = null
+
+    @Bindable
+    fun getName(): String = userModel?.name ?: ""
```
[the remark this answers is not in the document]
Comment (IstrajI) [reply app/src/main/java/pro/itshark/moneysplitter/presentation/userProfile/UserProfileViewModel.kt]: Not outdated

File: app/src/main/java/pro/itshark/moneysplitter/domain/UserInteractor.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package pro.itshark.moneysplitter.domain
+
+import io.reactivex.Single
+import pro.itshark.moneysplitter.model.repository.UserRepository
+import pro.itshark.moneysplitter.presentation.userProfile.UserModel
+
+class UserInteractor(private val userRepository: UserRepository) : UserUseCases {
+
+    override fun getUserInfo(): Single<UserModel> = userRepository.getUserInfo()
+            .map { user ->
```
[the remark this answers is not in the document]
Comment (BlaizMiko) [reply app/src/main/java/pro/itshark/moneysplitter/domain/UserInteractor.kt]: +

File: app/src/main/java/pro/itshark/moneysplitter/presentation/userProfile/UserProfileViewModel.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (BlaizMiko) [reply app/src/main/java/pro/itshark/moneysplitter/presentation/userProfile/UserProfileViewModel.kt]: +

File: app/src/main/java/pro/itshark/moneysplitter/di/AppComponent.kt [hunk 5]
```diff
@@ -1,11 +1,24 @@
...
+import dagger.BindsInstance
 import dagger.Component
 import pro.itshark.moneysplitter.di.modules.NetworkModule
+import pro.itshark.moneysplitter.di.modules.UserProfileModule
+import pro.itshark.moneysplitter.presentation.userProfile.UserProfileActivity
 import javax.inject.Singleton
 
 @Singleton
-@Component(modules = [NetworkModule::class])
+@Component(modules = [NetworkModule::class, UserProfileModule::class])
```
[the remark this answers is not in the document]
Comment (BlaizMiko) [reply app/src/main/java/pro/itshark/moneysplitter/di/AppComponent.kt]: (+)

**Comment by IstrajI** [comment]
You asked me to review changes, but there is no new commits, MB you forget to push them.

Also, when they would be done, mark discussions as resolved

**Pull request merged**